packages/connection/src/Manager.php

```diff
@@ -69,8 +69,31 @@ public function is_user_connected( $user_id ) {
 	 * @param Integer $user_id the user identifier.
 	 * @return Object the user object.
 	 */
-	public function get_connected_user_data( $user_id ) {
-		return $user_id;
+	public function get_connected_user_data( $user_id = null ) {
+		if ( ! $user_id ) {
+			$user_id = get_current_user_id();
+		}
+
+		$transient_key = "jetpack_connected_user_data_$user_id";
+
+		if ( $cached_user_data = get_transient( $transient_key ) ) {
+			return $cached_user_data;
+		}
+
+		\Jetpack::load_xml_rpc_client();
+		$xml = new \Jetpack_IXR_Client(
+			array(
+				'user_id' => $user_id,
+			)
+		);
+		$xml->query( 'wpcom.getUser' );
+		if ( ! $xml->isError() ) {
+			$user_data = $xml->getResponse();
+			set_transient( $transient_key, $xml->getResponse(), DAY_IN_SECONDS );
```

Comment:
Is `DAY_IN_SECONDS` going to work without a `\`?

Reply:
tested and it doesn't throw any errors

Reply:
Magnifique 👨‍🍳